### DISCUSSION TITLE
Combat, when the hero has a magic book but does not know any combat spells, the button for casting spells should be grayed out

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#10302] [General]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

In fheroes2, when the hero has a magic book but does not know any combat spells, the button for casting spells should be grayed out.
This way the player would not try to cast a spell and get a message window that he can't.

In fheroes2:

![Magic Book but no combat spell 001](https://user-images.githubusercontent.com/43583994/180643373-874899d4-0c5d-4ec2-a2d0-d090512db653.png)

![Magic Book but no combat spell 002](https://user-images.githubusercontent.com/43583994/180643375-2665aae0-5608-4b51-ae9e-eb4e5a5fd6f9.png)

![Magic Book but no combat spell 003](https://user-images.githubusercontent.com/43583994/180643379-4ee53e25-ccd1-4c48-bbde-c6ceced9a435.png)

### Describe the possible solution

It should be done in the same way that the hero can't access the magic book during a combat when he doesn't have one.

In fheroes2:

![Magic Book but no combat spell 004](https://user-images.githubusercontent.com/43583994/180643473-6066bbcb-d376-4f8b-a20a-d8344ad7f873.png)

**Comment by Alucard648** [comment]
I think, it`s fins as is.

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu 
I think the current implementation is better.
Players can distinguish two different situations: when a hero doesn't have a book at all and when a hero doesn't have a spells in a book.
If the button would be grayed, player could think, he doesn't have a book, which could force him to ignore shrines and other objects related to spells.

**Comment by LeHerosInconnu** [comment]
Hello @Alucard648 and @Branikolog,

> I think, it`s fins as is.

> Hi, @LeHerosInconnu I think the current implementation is better. Players can distinguish two different situations: when a hero doesn't have a book at all and when a hero doesn't have a spells in a book. If the button would be grayed, player could think, he doesn't have a book, which could force him to ignore shrines and other objects related to spells.

That's one point of view. :)
Personally, I'd rather not be able to click on a button that doesn't do what it's normally used for. :)

**Comment by ihhub** [comment]
Moving this issue to the Discussion section as we haven't found agreement.
